src/Features/VisualBasic/Portable/ConvertLinq/VisualBasicConvertLinqQueryToLinqMethodProvider.vb

```diff
@@ -0,0 +1,234 @@
+                    Case SyntaxKind.TakeClause
+                        Return CreateInvocationExpression(expression, NameOf(Enumerable.Take), DirectCast(queryClause, PartitionClauseSyntax).Count)
+                    Case SyntaxKind.OrderByClause
+                        Return ProcessOrderByClause(expression, DirectCast(queryClause, OrderByClauseSyntax))
+                    Case SyntaxKind.SelectClause
+                        Return ProcessSelectClause(expression, DirectCast(queryClause, SelectClauseSyntax))
+                    Case SyntaxKind.GroupJoinClause
+                        Return Nothing ' Too complicated for query method syntax.
+                    Case SyntaxKind.SimpleJoinClause
+                        Return Nothing ' Too complicated for query method syntax.
+                    Case SyntaxKind.LetClause
+                        Return Nothing ' Too complicated for query method syntax.
+                    Case SyntaxKind.AggregateClause
+                        Return Nothing ' Too complicated for query method syntax.
+                    Case SyntaxKind.GroupByClause
+                        Return Nothing ' Group by in query and in method are not 100% equivalent. Consdier support of some group by in the future.
+                    Case Else
+                        Return Nothing
+                End Select
+            End Function
+
+            Private Function ProcessFromClause(parentExpression As ExpressionSyntax, fromClause As FromClauseSyntax) As ExpressionSyntax
+                Dim expression = parentExpression
+                Dim variables = fromClause.Variables
+                If variables.Count > 1 Then
+                    Return Nothing ' Do not support more than 1 variable. It complicates query methods.
+                End If
+
+                Dim variable = variables.First()
+                If parentExpression Is Nothing Then
+                    expression = variable.Expression
+
+                    If variable.AsClause IsNot Nothing Then
+                        Dim identifier = SyntaxFactory.IdentifierName(NameOf(Enumerable.Cast))
+                        Dim generic = SyntaxFactory.GenericName(
+                            identifier.Identifier,
+                            SyntaxFactory.TypeArgumentList(SyntaxFactory.SingletonSeparatedList(variable.AsClause.Type)))
+                        expression = SyntaxFactory.InvocationExpression(
+                            SyntaxFactory.MemberAccessExpression(
+                            SyntaxKind.SimpleMemberAccessExpression,
+                            expression,
+                            SyntaxFactory.Token(SyntaxKind.DotToken), generic),
+                            SyntaxFactory.ArgumentList())
+                    End If
+
+                    Return expression
+                End If
+
+                Dim anonymousFunctions = FindAnonymousFunctionsFromParentInvocationOperation(variable.Expression)
```

Comment:
> Dim anonymousFunctions = FindAnonymousFunctionsFromParentInvocationOperation(variable.Expression) [](start = 15, length = 98)

I think the code should be defensive. We expect to get two lambdas back, we should verify that and do appropriate recovery if something went wrong.